include/irr/asset/format/encodePixels.h

```diff
@@ -491,7 +491,7 @@ namespace irr { namespace video
     {
         uint8_t* pix = reinterpret_cast<uint8_t*>(_pix);
         for (uint32_t i = 0u; i < 3u; ++i)
-            pix[i] = _input[i]*255.;
+            pix[i] = static_cast<uint8_t>(core::clamp(_input[i]*255.,0.4,255.4));
```

Comment:
@AnastaZIuk the compiler emits a fuckton of warnings about loosing precision, also some conversions like this one, need clamping to ensure no wrap-around takes place.

Reply:
@devshgraphicsprogramming what do you mean by?
> ensure no wrap-around takes place

Reply:
@devshgraphicsprogramming should I use Jesus operator or static_cast?

Reply:
`static_cast` definitely, when you want a conversion to happen as if a constructor was called.

Reply:
> ensure no wrap-around takes place

Integer and Normalized Formats (UNORM,NORM) have a limited range, and its very important that let's say 2.f doesn't turn into 510 and then wrap around when written to a R8 format (needs to clamp to 255)

Reply:
anyway do a few and I'll tell you if you're doing it well